scm/github/repo.go

```diff
@@ -258,13 +258,20 @@ func (c *client) Update(u *library.User, r *library.Repo, hookID int64) error {
 	}
 
 	// send API call to update the webhook
-	_, _, err := client.Repositories.EditHook(ctx, r.GetOrg(), r.GetName(), hookID, hook)
+	_, resp, err := client.Repositories.EditHook(ctx, r.GetOrg(), r.GetName(), hookID, hook)
+
+	// track if webhook exists in GitHub; a missing webhook
+	// indicates the webhook has been manually deleted from GitHub
+	webhookExists := true
+	if resp.StatusCode == http.StatusNotFound {
+		webhookExists = false
+	}
 
 	if err != nil {
-		return err
+		return webhookExists, err
 	}
 
-	return nil
+	return webhookExists, nil
 }
```

Comment:
suggestion: we can just collapse this to `return webhookExists, err` and skip the error check.

Reply:
addressed